### ISSUE TITLE
Style/MutableConstant should detect "late" freeze

### ISSUE THREAD

**Issue by sos4nt** [rubocop-hq/rubocop#4406]
I have a hash with a default value assigned to a constant like this:

```ruby
ERROR_CODES = {
  100 => :FooError,
  200 => :BarError,
  300 => :BazError
}
ERROR_CODES.default = :GenericError
ERROR_CODES.freeze
```

Although the hash is frozen, RuboCop objects that is is not. This is because it is not frozen upon assignment. IMO, RuboCop should also detect a "late" `freeze`, as shown in the example.

**Comment by bbatsov** [comment]
That's relatively hard to do, so I don't think we should bother with it. Obviously you can just move the `freeze` in your code.

**Comment by sos4nt** [comment]
> Obviously you can just move the freeze in your code.

Sorry, I don't see how I could move that `freeze`.

**Comment by bbatsov** [comment]
`default` doesn't work for frozen hashes?

**Comment by sos4nt** [comment]
No, calling `default=` is considered a modification of the hash.

The only workaround I can think of is:
```ruby
ERROR_CODES = {
  100 => :FooError,
  200 => :BarError,
  300 => :BazError
}.tap { |hash| hash.default = :GenericError }.freeze
```
but I'd rather avoid that hack.

**Comment by marcandre** [comment]
In case you haven't considered, I like the following:

```
ERROR_CODES = Hash.new(:GenericError).merge!(
  100 => :FooError,
  200 => :BarError,
  300 => :BazError
).freeze
```

**Comment by sos4nt** [comment]
@marcandre nice one, that would work. Unfortunately Rubycop doesn't enforce `freeze` in that case, it only does so for literals.

**Comment by znz** [comment]
I use following style to avoid such offense.

```ruby
error_codes = {
  100 => :FooError,
  200 => :BarError,
  300 => :BazError
}
error_codes.default = :GenericError
ERROR_CODES = error_codes.freeze
```

**Comment by vitormd** [comment]
Though I don't believe it is a good idea to kinda "Monkey patch" the hash default value.
I would live it as `nil`
You could still do: 
```ruby
ERROR_CODES = {
  100 => :FooError,
  200 => :BarError,
  300 => :BazError
}
DEFAULT_ERROR = :GenericError
```
or
```ruby
ERROR_CODES = {
  100 => :FooError,
  200 => :BarError,
  300 => :BazError
}
```

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.

The problem seems to be pretty rare in practice, so probably just some exclusion would be fine for the people affected by it.
